### PR TITLE
Adding Bluecore integration

### PR DESCRIPTION
**Pull request by proModeLife**
## What are the changes introduced in this PR?
Adding bluecore integration.
Write a brief explainer on your code changes.
Followed standard procedure to add a destination and its test case. 
## What is the related Linear task?
https://linear.app/rudderstack/issue/INT-70/event-stream-destination-bluecore

resolves INT-70

Resolves INT-70

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: src/v0/destinations/bluecore/utils.js [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+};
+
+/**
+ *
+ * @param {*} message Rudder element
+ * @param {*} categoryToContent example: [ { from: 'clothing', to: 'product' } ]
+ *
+ * Handles order completed and checkout started types of specific events
+ */
+const handleOrder = (message, categoryToContent) => {
```
Comment (shrouti1507) [inline src/v0/destinations/bluecore/utils.js:55]: this implementation is specific to facebook, please edit according to bluecore needs

File: src/v0/destinations/bluecore/utils.js [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+    product_id: productId,
+    sku,
+    name,
+    brand,
+    variant,
+    price: priceString, // Renamed to avoid conflict with modified price
+    currency,
+    position,
+    url,
+    image_url: imageUrl,
```
Comment (shrouti1507) [inline src/v0/destinations/bluecore/utils.js:126]: where is this coming from

File: src/v0/destinations/bluecore/utils.js [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+ * @param {*} categoryToContent Example: [ { from: 'clothing', to: 'product' } ]
+ * @param {*} valueFieldIdentifier it can be either value or price which will be matched from properties and assigned to value for fb payload
+ */
+const handleProduct = (message, categoryToContent, valueFieldIdentifier) => {
+  const contentIds = [];
+  const products = [];
+  const useValue = valueFieldIdentifier === 'properties.value';
+  const contentId =
+    message.properties?.product_id || message.properties?.sku || message.properties?.id;
+  const contentType = getContentType(message, 'product', categoryToContent);
```
Comment (shrouti1507) [inline src/v0/destinations/bluecore/utils.js:171]: why is this needed

File: src/v0/destinations/bluecore/utils.js [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+    content_name: contentName,
+    content_category: getContentCategory(contentCategory),
+    currency,
+    value,
+    products,
+  };
+};
+
+const handleSearch = (message) => {
+  const query = message?.properties?.query;
```
Comment (shrouti1507) [inline src/v0/destinations/bluecore/utils.js:198]: this is specific to facebook pixel

File: src/v0/destinations/bluecore/utils.js [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+    contentIds.push(contentId);
+    products.push({
+      id: contentId,
+      quantity: message?.properties?.quantity || 1,
+      item_price: message?.properties?.price,
+    });
+  }
+  return {
+    content_ids: contentIds,
+    content_category: getContentCategory(contentCategory),
```
Comment (shrouti1507) [inline src/v0/destinations/bluecore/utils.js:226]: why is this required?

File: src/v0/destinations/bluecore/test.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+const eventProcessor = require('./transform');
```
Comment (shrouti1507) [inline src/v0/destinations/bluecore/test.js:1]: please remove this file

File: src/v0/destinations/bluecore/data/bluecoreTrackConfig.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
+[
+  {
```
Comment (shrouti1507) [inline src/v0/destinations/bluecore/data/bluecoreTrackConfig.json:2]: why customer information is coming in track call?

File: src/v0/destinations/bluecore/data/bluecoreTrackConfig.json [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    "required": false
+  },
+  {
+    "destKey": "customer.age",
+    "sourceKeys": "age",
+    "required": false
+  },
+  {
+    "destKey": "customer.sex",
+    "sourceKeys": "sex",
```
Comment (shrouti1507) [inline src/v0/destinations/bluecore/data/bluecoreTrackConfig.json:19]: same

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Comment by shrouti1507** [comment]
we are using https://github.com/rudderlabs/rudder-transformer/pull/3061 to continue on this. So closing this PR
